pkg/apis/resource/types.go

```diff
@@ -1183,6 +1245,16 @@ type AllocationResult struct {
 	// it got removed. May be reused once decoding v1alpha3 is no longer
 	// supported.
 	// Controller string
+
+	// AllocationTimestamp stores the time when the resources were allocated.
+	// This field is not guaranteed to be set, in which case that time is unknown.
+	//
+	// This is an alpha field and requires enabling the DRADeviceBindingConditions and DRAResourceClaimDeviceStatus
+	// feature gate.
+	//
+	// +optional
+	// +featureGate=DRADeviceBindingConditions,DRAResourceClaimDeviceStatus
```

Comment:
I could imagine use cases for this besides DRADeviceBindingConditions, so making it depend on that feature gate might be too strict.

But introducing a new field in this KEP with just `DRAResourceClaimDeviceStatus` as feature gate (not really the right one) or no feature gate at all (core DRA = structured parameters) would be odd. Let's keep it like this.